### ISSUE TITLE
[question]: Unclear behavior when not specifying is_android and is_ios flags

### ISSUE THREAD

**Issue by mochizuki-pg** [OneSignal/onesignal-ruby-api#33]
### How can we help?

when I tried sending notifications to users who are subscribed on both Android and iOS platforms, I noticed that unless I explicitly set the is_android and is_ios flags, notifications weren't sent to my Android users.

```ruby
OneSignal::Notification.new(
  app_id: ,
  contents: ,
  channel_for_external_user_ids: 'push',
  include_external_user_ids: ,
  app_url: ,
  ios_attachments: ,
  existing_android_channel_id: ,
  is_ios: true,    
  is_android: true 
)
```

documentation lists several flags like is_android, is_ios, is_huawei, is_any_web, is_chrome_web, is_firefox, is_safari, is_wp_wns, is_adm, and is_chrome as [optional]. However, the observed behavior suggests that we might need to specify them to ensure delivery. while these parameters are labeled as optional, do we essentially need to specify all of them to ensure the messages are delivered correctly?

The discrepancy between the documentation and the actual behavior has caused some confusion. Any clarification or guidance on this matter would be greatly appreciated.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bismark64** [comment]
Same issue here. Is this optional or required? What about if we don't want to specify to which platform we want to deliver?

**Comment by mochizuki-pg** [comment]
@bismark64 
Perhaps it is better to specify `true` or `false` explicitly, although the documentation mentions `optional`

**Comment by mochizuki-pg** [comment]
For iOS only, I could send without writing
When adding other devices, such as Android, it was not sent to Android unless I explicitly wrote "true".

**Comment by bismark64** [comment]
Yeah docs are really not clear regarding this. It's not optional if without it, it does not work imo.. 
It works for ios because `is_ios` defaults to `true` in any request. But if you want any other device, apparently, you need to explicitly define it.

**Comment by sherwinski** [comment]
Hey everyone,

In [v5.3.0](https://github.com/OneSignal/onesignal-ruby-api/releases/tag/v5.3.0), the targeting model has been updated: `include_external_user_ids` and `channel_for_external_user_ids` have been replaced by `include_aliases` and `target_channel`. Here's an example:

```ruby
OneSignal::Notification.new(
  app_id: 'YOUR_APP_ID',
  contents: { en: 'Hello!' },
  include_aliases: { 'external_id' => ['user_1', 'user_2'] },
  target_channel: 'push'
)
```

Regarding the `is_ios` / `is_android` flags — these are optional platform filters that allow you to restrict which platforms receive the notification. By default, if none are set, the notification will target all platforms. However, setting platform-specific content (like `ios_attachments`) without explicitly opting in other platforms can affect delivery. For more details, see the [Push API reference](https://documentation.onesignal.com/reference/push-notification) and [Push Templates reference](https://documentation.onesignal.com/reference/create-template#push-templates).

Closing this as the v5 library introduces a clearer targeting model. If you're still seeing unexpected delivery behavior on v5, please open a new issue with specifics.
